### DISCUSSION TITLE
SUEnableAutomaticChecks in Info.plist not respected after update?

### DISCUSSION THREAD

**Discussion by coryl** [sparkle-project/Sparkle#2487] [Q&A] [answered]
My Info.plist looks like this:
```
<key>SUEnableAutomaticChecks</key>
<true/>
<key>SUAllowsAutomaticUpdates</key>
<true/>
<key>SUAutomaticallyUpdate</key>
<true/>
```

On 1st app launch, it looks like these properties are respected: an app update is downloaded in the background, and on quit the app is updated. At this point, the only keys set in preferences plist is `SULastCheckTime`, `SUHasLaunchedBefore`, `SUUpdateGroupIdentifier`.

On 2nd app launch (the app is now up to date), the user receives the "Check for updates automatically?" dialog box. Responding to this box sets additional keys in the preference plist, according to the user's response.

So my understanding was that by setting the `SUEnableAutomaticChecks` value in the Info.plist, we would have avoided this dialog box forever. However it seems that on second launch (or after an app update?), this Info.plist flag is no longer recognized.

Am I misunderstanding these flags? Or is this an edge case? Thanks

**Comment by coryl** [comment]
Ok I think I understand now.

Some Info.plist customizations are only INITIAL values that will no longer be read/respected if the user makes setting changes via the default permissions dialog box.

To set customizations, we should use properties/functions on the updater object itself, or possibly delegate implementations.

**Comment by zorgiepoo** [comment, marked as answer]
> On 2nd app launch (the app is now up to date), the user receives the "Check for updates automatically?" dialog box. Responding to this box sets additional keys in the preference plist, according to the user's response.

> So my understanding was that by setting the SUEnableAutomaticChecks value in the Info.plist, we would have avoided this dialog box forever. However it seems that on second launch (or after an app update?), this Info.plist flag is no longer recognized.

The permission prompt only shows up if `SUEnableAutomaticChecks` is not set in the Info.plist. If you want the default behavior where Sparkle asks the user permission to check for updates automatically, do not set `SUEnableAutomaticChecks` in the Info.plist. Also if you do not set `SUEnableAutomaticChecks` in the Info.plist, Sparkle should not check for updates or download updates before asking your user permission for checking for updates.

Your second build after your update was probably configured differently than the first build you tested.

> Some Info.plist customizations are only INITIAL values that will no longer be read/respected if the user makes setting changes via the default permissions dialog box.

Info.plist settings are indeed for initial values and users can change those options from permissions dialog box, or from your app's settings if your app exposes settings for them.

> To set customizations, we should use properties/functions on the updater object itself, or possibly delegate implementations.

Usually you only want to do this if your app exposes UI options for this like in your app's settings. Do not set these kinds of properties on the updater object randomly / all the time / every launch / etc.

**Reply by coryl** [reply]
Thanks. The thing that was affecting my learning the most was that the original build uploaded to a server was indeed different. I got caught up manipulating settings in Xcode and then running it locally, but forgot the version uploaded to the web has different settings.
